### PR TITLE
Solve3sum sol

### PR DESCRIPTION
**Pull request by HarryWangATX**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

**Comment by HarryWangATX** [comment]
😭😭😭

File: solutions/usaco-994.mdx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+public class Threesum {
+	public static void main(String[] args) throws IOException {
+		Kattio io = new Kattio("threesum");
+
+		int N = io.nextInt();
+		int Q = io.nextInt();
+		int[] a = new int[N];
+		long[][] dp = new long[N + 1][N + 1];
+		int[] cnt = new int[2 * 1000000 + 1];
```
Comment (SansPapyrus683) [inline solutions/usaco-994.mdx:30]: put 1e6 in a variable like MAX_VAL or smth

File: solutions/usaco-994.mdx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+				if(k >= -1000000 && k <= 1000000) {
+					dp[i + 1][j + 1] += cnt[k + 1000000];
+				}
+
+				cnt[a[j] + 1000000]++;
+			}
+		}
+		
+		for (int i = N; i >= 1; i--) {
+			for (int j = i+1; j <= N; j++) {
```
Comment (SansPapyrus683) [inline solutions/usaco-994.mdx:52]: spaces

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
actually pretty good

**Review by SansPapyrus683 — APPROVED** [review]

File: solutions/usaco-994.mdx [hunk 3]
```diff
@@ -0,0 +1,71 @@
+---
+id: usaco-994
+source: USACO Gold 2020 January
+title: Farmer John Solves 3SUM
+author: Qi Wang
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_threesum_gold_jan20.html)
```
Comment (ryanchou-dev) [inline solutions/usaco-994.mdx:8]:
```suggestion
[Official Analysis (C++ and Java)](http://www.usaco.org/current/data/sol_threesum_gold_jan20.html)
```

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]
add comments as well

**Review by ryanchou-dev — APPROVED** [review]
⭐

**Pull request merged**